From 8827dc27a4507ece7b2b84e1b28eb6ae532fda1d Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Wed, 15 Apr 2015 11:55:19 +0530
Subject: [PATCH] OPREGION: Fixing the error in the dptf pen function

Issue description: Previous implementation used to return bad parameter
for dptf pen when called with value 0 or 1 or any other value. Expected
result is true in case 0 or 1 and bad parameter in case of values other
than 0 or 1.

Solution: Corrected the check to validate the value appropriately and
return true in case the value is 0 or 1 and false otherwise.

Change-Id: I7a26d5d736d268963d16265d69f81f418a11c6f8
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1411
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
---
 drivers/acpi/pmic_opregion/intel_soc_pmic_opregion.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/acpi/pmic_opregion/intel_soc_pmic_opregion.c b/drivers/acpi/pmic_opregion/intel_soc_pmic_opregion.c
index fa5e6c4..a703a96 100644
--- a/drivers/acpi/pmic_opregion/intel_soc_pmic_opregion.c
+++ b/drivers/acpi/pmic_opregion/intel_soc_pmic_opregion.c
@@ -237,7 +237,7 @@ static acpi_status pmic_dptf_pen(struct intel_soc_pmic_opregion *opregion,
 	if (function == ACPI_READ)
 		return d->get_policy(reg, value) ? AE_ERROR : AE_OK;
 
-	if (*value != 0 || *value != 1)
+	if ((*value != 0) && (*value != 1))
 		return AE_BAD_PARAMETER;
 
 	return d->update_policy(reg, *value) ? AE_ERROR : AE_OK;
-- 
1.9.1

